tools/gokvm-caps.go

```diff
@@ -0,0 +1,246 @@
+		kvm.CapEnableCap,
+		kvm.CapPPCGetPVInfo,
+		kvm.CapPPCIRQLevel,
+		kvm.CapPPCBookeSREGS,
+		kvm.CapSPAPRTCE,
+		kvm.CapPPCSMT,
+		kvm.CapPPCRMA,
+		kvm.CapPPCHIOR,
+		kvm.CapPPCPAPR,
+		kvm.CapSWTLB,
+		kvm.CapONEREG,
+		kvm.CapPPCGetSMMUInfo,
+		kvm.CapPPCAllocHTAB,
+		kvm.CapPPCBokkeWatchdog,
+		kvm.CapPPCHTABFD,
+		kvm.CapPPCEPR,
+		kvm.CapDeviceCtrl,
+		kvm.CapPPCRTAS,
+		kvm.CapPPCFixupHCALL,
+		kvm.CapPPCEnableHCALL,
+		kvm.CapPPCHWRNG,
+		kvm.CapSPAPRTCE64,
+		kvm.CapVCPUAttributes,
+		kvm.CapPPCHTM,
+		kvm.CapSPAPRResizeHPT,
+		kvm.CapPPCMMURADIX,
+		kvm.CapPPCMMUHASHV3,
+		kvm.CapPPCFWNMI,
+		kvm.CapPPCSMTPossible,
+		kvm.CapPPCGetCPUChar,
+		kvm.CapPPCNestedHV,
+		kvm.CapCoalescedPIO,
+		kvm.CapPPCIRQXive,
+		kvm.CapPPCGuestDebugSStep,
+		kvm.CapPPCSecureGuest,
+		kvm.CapPPCDAWR1,
+		kvm.CapPPCRPTInvalidate,
+		kvm.CapBinaryStatsFD,
+		kvm.CapPPCAILMode3,
+	}
+
+	S390tests := []kvm.Capability{
+		kvm.CapUserMemory,
+		kvm.CapMPState,
+		kvm.CapCoalescedMMIO,
+		kvm.CapSetGuestDebug,
+		kvm.CapIRQRouting,
+		kvm.CapIRQFD,
+		kvm.CapIOEventFD,
+		kvm.CapS390PSW,
+		kvm.CapEnableCap,
+		kvm.CapONEREG,
+		kvm.CapS390GMap,
+		kvm.CapS390UControl,
+		kvm.CapS390COW,
+		kvm.CapS390CSSSupport,
+		kvm.CapDeviceCtrl,
+		kvm.CapS390IRQCHIP,
+		kvm.CapVMAttributes,
+		kvm.CapS390UserSIGP,
+		kvm.CapS390VectorRegisters,
+		kvm.CapS390MemOp,
+		kvm.CapS390UserSTSI,
+		kvm.CapS390SKEYS,
+		kvm.CapS390InjectIRQ,
+		kvm.CapS390IRQState,
+		kvm.CapS390RI,
+		kvm.CapVCPUAttributes,
+		kvm.CapS390UserINSTR0,
+		kvm.CapS390GS,
+		kvm.CapS390AIS,
+		kvm.CapS390CMMAMigration,
+		kvm.CapS390AISMigration,
+		kvm.CapS390BPB,
+		kvm.CapS390HPage1M,
+		kvm.CapCoalescedPIO,
+		kvm.CapS390VCPUResets,
+		kvm.CapS390Protected,
+		kvm.CapS390DIAG318,
+		kvm.CapBinaryStatsFD,
+		kvm.CapSysAttributes,
+		kvm.CapS390MemOpExtention,
+		kvm.CapS390ProtectedDump,
+		kvm.CapS390ZPCIOP,
+		kvm.CapS390CPUTOPOLOGY,
+	}
+
+	ARM64tests := []kvm.Capability{
+		kvm.CapIRQChip,
+		kvm.CapUserMemory,
+		kvm.CapMPState,
+		kvm.CapCoalescedMMIO,
+		kvm.CapSetGuestDebug,
+		kvm.CapIRQRouting,
+		kvm.CapIRQFD,
+		kvm.CapIOEventFD,
+		kvm.CapVCPUEvents,
+		kvm.CapINTRShadow,
+		kvm.CapONEREG,
+		kvm.CapSignalMSI,
+		kvm.CapARMPSCI,
+		kvm.CapARMSetDeviceAddr,
+		kvm.CapDeviceCtrl,
+		kvm.CapARMEL132BIT,
+		kvm.CapVMAttributes,
+		kvm.CapARMPSCI02,
+		kvm.CapARMPMUV3,
+		kvm.CapVCPUAttributes,
+		kvm.CapARMUserIRQ,
+		kvm.CapARMInjectSErrorESR,
+		kvm.CapCoalescedPIO,
+		kvm.CapManualDirtyLogProtect2,
+		kvm.CapARMVMIPASize,
+		kvm.CapARMSVE,
+		kvm.CapARMPTRAuthAddress,
+		kvm.CapARMPTRAuthGeneric,
+		kvm.CapARMIRQLineLayout2,
+		kvm.CapARMNISVToUser,
+		kvm.CapARMInjectEXTDABT,
+		kvm.CapBinaryStatsFD,
+		kvm.CapARMMTE,
+		kvm.CapSysAttributes,
+		kvm.CapARMSystemSuspend,
+	}
+
+	kvmFile, err := os.Open("/dev/kvm")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	kvmfd := kvmFile.Fd()
+	tres := make(results, 0)
+
+	var runtests []kvm.Capability
+
+	switch runtime.GOARCH {
```

Comment:
Since gokvm currently only supports amd64, could you remove the code for other architectures?
If we support other architectures, I'd like to split the file and distinguish by build constraints like `//go:build amd64`.

https://pkg.go.dev/cmd/go#hdr-Build_constraints

Reply:
removed it

Reply:
Thanks!